criu/uffd.c

```diff
@@ -1466,20 +1466,28 @@ int cr_lazy_pages(bool daemon)
 	 */
 	nr_fds = task_entries->nr_tasks + (opts.use_page_server ? 2 : 1);
 	epollfd = epoll_prepare(nr_fds, &events);
-	if (epollfd < 0)
+	if (epollfd < 0) {
+		if (events)
+			xfree(events);
```

Comment:
Looks like double free. On error epoll_prepare already frees events, isn't it?

Reply:
Yes, I guess the confusion here is in `epoll_prepare()` it is compared to '-1' and here it is '< 0'. Reading the man-page this is correct, but probably something coverity does not know about.